File: 88-merge-sorted-array/merge-sorted-array.go
func merge(nums1 []int, m int, nums2 []int, n int) {
	if n == 0 {
		return
	}
	if m == 0 {
		for i := 0; i < n; i++ {
			nums1[i] = nums2[i]
		}
		return
	}
	arr1 := m - 1
	arr2 := n - 1
	curr := m + n - 1

	for arr2 >= 0 {
		if arr1 >= 0 && nums1[arr1] > nums2[arr2] {
			nums1[curr] = nums1[arr1]
			arr1--
		} else {
			nums1[curr] = nums2[arr2]
			arr2--
		}
		curr--
	}
}

File: 88-merge-sorted-array/merge-sorted-array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        
        index1 = m - 1
        index2 = n - 1
        index = m + n - 1
        while index1 >= 0 and index2 >= 0:
            if (nums1[index1] > nums2[index2]):
                nums1[index] = nums1[index1]
                index1 -= 1
                index -= 1
            else:
                nums1[index] = nums2[index2]
                index2 -= 1
                index -= 1
        
        if index1 != -1:
            while index1 >= 0:
                nums1[index] = nums1[index1]
                index1 -= 1
                index -= 1
        else:
             while index2 >= 0:
                nums1[index] = nums2[index2]
                index2 -= 1
                index -= 1
        
    

